### ISSUE TITLE
All nodes offline on ci-release.nodejs.org

### ISSUE THREAD

**Issue by targos** [nodejs/build#3177]
The reason Jenkins gives is "This node is offline because it uses an old agent.jar"

Note that it happened after a Jenkins restart that I did after updating a couple of plugins which had security vulnerabilities. I **did not** update Jenkins itself.

@nodejs/build-infra

**Comment by richardlau** [comment]
> The reason Jenkins gives is "This node is offline because it uses an old agent.jar"
> 
> Note that it happened after a Jenkins restart that I did after updating a couple of plugins which had security vulnerabilities. I **did not** update Jenkins itself.

ugh, that's suprising. Our playbooks download the agent jar via https://github.com/nodejs/build/blob/46a768e15303ab3c8b94fa7ddfaf33da07db1c9e/ansible/roles/jenkins-worker/tasks/main.yml#L67-L74
We can try reansibling one of the machines. Unfortunately that will be the full [`jenkins/worker/create.yml` playbook](https://github.com/nodejs/build/blob/main/ansible/playbooks/jenkins/worker/create.yml) as 
[`jenkins/worker/upgrade-jar.yml`](https://github.com/nodejs/build/blob/main/ansible/playbooks/jenkins/worker/upgrade-jar.yml) only works on the test hosts (I don't know what the reason for https://github.com/nodejs/build/blob/46a768e15303ab3c8b94fa7ddfaf33da07db1c9e/ansible/playbooks/jenkins/worker/upgrade-jar.yml#L7 is).

**Comment by targos** [comment]
> Our playbooks download the agent jar via

Maybe that's the problem somehow?

Release agents download the agent jar using `ci.nodejs.org` which is on an older Jenkins version.

https://github.com/nodejs/build/blob/46a768e15303ab3c8b94fa7ddfaf33da07db1c9e/ansible/group_vars/release.yml#L2-L3

**Comment by richardlau** [comment]
Ah yeah, that is going to be a problem 😞.

**Comment by targos** [comment]
Something is really wrong. I updated the jar on `release-nearform-macos10.15-x64-1` but it still can't connect.

```diff
diff --git a/ansible/group_vars/release.yml b/ansible/group_vars/release.yml
index 1c9b53a..b6311d1 100644
--- a/ansible/group_vars/release.yml
+++ b/ansible/group_vars/release.yml
@@ -1,6 +1,5 @@
 jenkins_url: "https://ci-release.nodejs.org"
-# intentionally fetching `slave.jar` from ci.nodejs.org to avoid auth problems
-jenkins_worker_jar: "https://ci.nodejs.org/jnlpJars/slave.jar"
+jenkins_worker_jar: "{{ jenkins_url }}/jnlpJars/slave.jar"
 jenkins_icon: "https://ci.nodejs.org/favicon.ico"
 server_user: "iojs"
 home: "/home"
diff --git a/ansible/playbooks/jenkins/worker/upgrade-jar.yml b/ansible/playbooks/jenkins/worker/upgrade-jar.yml
index 1868eb8..e11fe23 100644
--- a/ansible/playbooks/jenkins/worker/upgrade-jar.yml
+++ b/ansible/playbooks/jenkins/worker/upgrade-jar.yml
@@ -4,7 +4,9 @@
 # upgrade `slave.jar` at jenkins workers
 #

--  hosts: test
+-  hosts:
+    - test
+    - release

    tasks:
     - name: download slave.jar
```

**Comment by targos** [comment]
So, I clicked on "Bring this node back online" on all the available nodes and it seems to have worked?

There are still quite a few offline nodes, though (for other reasons)

**Comment by targos** [comment]
Only `release-nearform-macos10.15-x64-1` manually updated has a remoting version of `3071.v7e9b_0dc08466`. It's probably a good idea to update all of them.

**Comment by targos** [comment]
I finally understand what happened.

Before updating the plugins (and restarting), I installed the [Versions Node Monitors plugin](https://plugins.jenkins.io/versioncolumn). This plugin is already installed on ci.nodejs.org and is useful to know what version of Java is installed on each node just by looking at the table in https://ci-release.nodejs.org/computer/
It happens that the plugin has a default configuration that will disconnect agents which have a lower Java version than the main node and that do not have the same version of the agent jar.

I disabled these options:
<img width="856" alt="CleanShot 2023-02-05 at 15 10 26@2x" src="https://user-images.githubusercontent.com/2352663/216824666-d0bda7a3-c622-44ea-87a3-28698fb5a160.png">
